### PR TITLE
Export progress

### PR DESCRIPTION
**Pull request by mknblch**
- moved the export logic from several classes into ParallelExporter
- created generic exporters usable for several algorithms (instead of one exp. for each algo)
- added ProgressLogging logic to ParallelExporter
- add progress logger testcase

### CODE REVIEW

File: tests/src/test/java/org/neo4j/graphalgo/algo/ShortestPathsProcTest.java [hunk 1]
```diff
@@ -123,6 +123,8 @@ public void testResultStream() throws Exception {
             return true;
         });
 
+        System.out.println();
```
Comment (jexp) [inline tests/src/test/java/org/neo4j/graphalgo/algo/ShortestPathsProcTest.java]: why do we need these?

File: core/src/main/java/org/neo4j/graphalgo/core/utils/ParallelExporter.java [hunk 2]
```diff
@@ -1,33 +1,118 @@
...
+import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class ParallelExporter<T> extends Exporter<T> {
+/**
+ * @author mknblch
+ */
+public abstract class ParallelExporter<T> extends AbstractExporter<T> {
 
-    private final ExecutorService executor;
+    public static final int DEFAULT_CONCURRENCY = Runtime.getRuntime().availableProcessors() * 2;
```
Comment (jexp) [inline core/src/main/java/org/neo4j/graphalgo/core/utils/ParallelExporter.java]: we should move this into some central place, 
also make concurrency configurable

File: core/src/main/java/org/neo4j/graphalgo/core/utils/ParallelExporter.java [hunk 3]
```diff
@@ -1,33 +1,118 @@
...
-            ExecutorService executor) {
-        super(api);
-        this.batchNodes = batchNodes;
-        this.executor = executor;
-        this.batchSize = batchSize;
+    public static final String TASK_EXPORT = "EXPORT";
+
+    private final ExecutorService executorService;
+
+    private int concurrency = DEFAULT_CONCURRENCY;
```
Comment (jexp) [inline core/src/main/java/org/neo4j/graphalgo/core/utils/ParallelExporter.java]: take concurrency from procedure config via fluent setter

**Comment by jexp** [comment]
Good call, I had some small comments regarding setting the concurrency, but I'll merge this anyway.

File: tests/src/test/java/org/neo4j/graphalgo/algo/ShortestPathsProcTest.java (hunk 1 above)
Comment (mknblch) [inline tests/src/test/java/org/neo4j/graphalgo/algo/ShortestPathsProcTest.java]: woops. forget to remove. i was testing load/eval/write progress log and needed a newline ;)

**Comment by mknblch** [comment]
im going to fix them in a followup branch

**Pull request merged**